linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -135,6 +136,25 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     }
   }
 
+  private def checkJSSuperClass(classDef: LinkedClass): Unit = {
+    implicit val ctx = ErrorContext(classDef)
+
+    if (classDef.kind.isJSClass) {
+      classDef.jsSuperClass.fold {
+        // .get is OK: the Analyzer checks that a super class is present.
+        val superClass = lookupClass(classDef.superClass.get.name)
+        if (superClass.jsClassCaptures.isDefined)
+          reportError(i"super class ${superClass.name} may not have jsClassCaptures")
+      } { tree =>
+        val env = Env.fromSignature(AnyType, classDef.jsClassCaptures, Nil, AnyType)
```

Comment:
I think `thisType` should be `NoType` instead of `AnyType`, shouldn't it?
```suggestion
        val env = Env.fromSignature(NoType, classDef.jsClassCaptures, Nil, AnyType)
```

Reply:
Ah, yes, of cousre.